src/logic/safe/store/selectors/gatewayTransactions.ts

```diff
@@ -166,3 +166,52 @@ export const getLastTransaction = createSelector(
 export const getLastTxNonce = createSelector(getLastTransaction, (lastTx) => {
   return isMultisigExecutionInfo(lastTx?.executionInfo) ? lastTx?.executionInfo.nonce : undefined
 })
+
+export const getBatchableTransactions = createSelector(
+  nextTransaction,
+  queuedTransactions,
+  currentSafeNonce,
+  (nextTxs, queuedTxs, safeNonce) => {
+    const batchableTransactions: Transaction[] = []
+    let currentNonce = safeNonce
+
+    if (!nextTxs) return batchableTransactions
+
+    function isTxEligible(transaction: Transaction): boolean {
+      return (
+        isMultisigExecutionInfo(transaction.executionInfo) &&
+        transaction.executionInfo.nonce === currentNonce &&
+        transaction.executionInfo.confirmationsSubmitted >= transaction.executionInfo.confirmationsRequired
+      )
+    }
+
+    function addToBatchIfEligible(transaction: Transaction): void {
+      if (
+        isMultisigExecutionInfo(transaction.executionInfo) &&
+        isTxEligible(transaction) &&
+        batchableTransactions.length < BATCH_LIMIT
+      ) {
+        batchableTransactions.push(transaction)
+        currentNonce = transaction.executionInfo.nonce + 1
+      }
+    }
+
+    const eligibleTransactions = nextTxs.filter(isTxEligible)
+
+    for (const eligibleTransaction of eligibleTransactions) {
+      addToBatchIfEligible(eligibleTransaction)
+    }
+
+    if (!queuedTxs) return batchableTransactions
+
+    Object.values(queuedTxs).forEach((queuedTxsByNonce) => {
+      const eligibleTransactions = queuedTxsByNonce.filter(isTxEligible)
+
+      for (const eligibleTransaction of eligibleTransactions) {
+        addToBatchIfEligible(eligibleTransaction)
+      }
+    })
```

Comment:
If you looped over next and queued transactions at the same time, you wouldn't have to separate the logic here.

Reply:
Good suggestion! Combining them works well